### PR TITLE
Add Arabic localizations to Sileo

### PR DESCRIPTION
**Pull request by worldlydev**
Original PR #20. Closed due to rebasing. Credit to @iA7myd

### CODE REVIEW

**Comment by worldlydev** [comment]
Needs verification of proper RTL language support as mentioned in #33

**Comment by CharlieWhile13** [comment]
Let me get these verified with some other people, and I'll look into support/merging 👍

**Comment by worldlydev** [comment]
> Let me get these verified with some other people, and I'll look into support/merging 👍

Definitely missing strings. I'm adding them in now and reformatting to match base. Shouldn't take more than a couple minutes.

Somebody else will have to translate the newly added strings though.

**Comment by CharlieWhile13** [comment]
Sure. I'll get that sorted too

**Review by yazanoo16 — CHANGES_REQUESTED** [review]
the first string could say "بدون فئة"
the third string could say "السمات"
and the tenth string could say "دعم المحطّة الطرفية"

everything else seems to be good so far

**Comment by worldlydev** [comment]
> everything else seems to be good so far

Please be more specific than first, second, and third string. I'm not an Arabic speaker so I'm not sure which strings you are referring to.

**Comment by worldlydev** [comment]
**STRINGS TO LOOK AT/NEED TRANSLATING**

_In Localizable.strings_

_Possibly correct:_
"__LocalizationCredit.Title" = "ترجمه إلى العربية iA7myd.";

_Need translating:_
"Theme" = "Theme";
"Tint_Color" = "Tint Color";
"Reset_Tint_Color" = "Reset Tint Color";
"Auto_Complete_Queue" = "Auto Complete Queue";
"Enable_Analytics" = "Enable Analytics";

"Sileo_Light" = "Sileo Light";
"Sileo_Dark" = "Sileo Dark";
"Sileo_Adaptive" = "Sileo Adaptive";
"System" = "System";

"View_Depiction" = "View Depiction";

**Comment by worldlydev** [comment]
All strings are added and structured the same as base.lproj 

Just needs a little translation input and it should be ready to go :)

File: Sileo/Sileo/ar.lproj/Errors.strings [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+"Error_Indicator" = "خطأ: %@";
+
+"No_Package.Title" = "تعذر العثور على حزمة %@ في المستودعات المثبتة لديك حاليًا.";
+"No_Package.Body" = "إذا كان جديدًا، فانتظر حتى ينتهي Sileo من التحديث، ثم حاول مرة أخرى. تحقق عبر الإنترنت للمزيد من المساعدة.";
+
+"Invalid_URL.Title" = "عنوان URL غير صالح";
+
+"Download_Size_Mismatch" = "الحجم المتوقع %@، تم الحصول على %@ بدلًا من ذلك.";
+"Download_Failing_Status_Code" = "فشل بالحالة %ld";
+"Download_Hash_Mismatch" = "التجزئة المتوقعة %@، تم الحصول على %@ بدلًا من ذلك.";
```
Comment (yazanoo16) [inline Sileo/Sileo/ar.lproj/Errors.strings:21]: this one could also say

"Download_Hash_Mismatch" = "التجزئة المتوقعة هي %@، لكن تم الحصول على %@ بدلًا من ذلك.";
“

**Review by yazanoo16 — CHANGES_REQUESTED** [review]

**Comment by yazanoo16** [comment]
> > everything else seems to be good so far
> 
> Please be more specific than first, second, and third string. I'm not an Arabic speaker so I'm not sure which strings you are referring to.

i was basically referring to String 1, String 3 and String 10

File: Sileo/Sileo/ar.lproj/Errors.strings [hunk 2]
```diff
@@ -0,0 +1,49 @@
+"Unknown" = "حدث خطأ غير معروف.";
+"Error_Indicator" = "خطأ: %@";
+
+"No_Package.Title" = "تعذر العثور على حزمة %@ في المستودعات المثبتة لديك حاليًا.";
+"No_Package.Body" = "إذا كان جديدًا، فانتظر حتى ينتهي Sileo من التحديث، ثم حاول مرة أخرى. تحقق عبر الإنترنت للمزيد من المساعدة.";
```
[the remark this answers is not in the document]
Comment (worldlydev) [reply Sileo/Sileo/ar.lproj/Errors.strings]: Fixed

File: Sileo/Sileo/ar.lproj/Errors.strings [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+"Invalid_URL.Title" = "عنوان URL غير صالح";
+
+"Download_Size_Mismatch" = "الحجم المتوقع %@، تم الحصول على %@ بدلًا من ذلك.";
```
[the remark this answers is not in the document]
Comment (worldlydev) [reply Sileo/Sileo/ar.lproj/Errors.strings:19]: Fixed

File: Sileo/Sileo/ar.lproj/Errors.strings (hunk 1 above)
Comment (worldlydev) [reply Sileo/Sileo/ar.lproj/Errors.strings]: Fixed

**Review by iA7myd — APPROVED** [review]

**Comment by iA7myd** [comment]
"Theme" = "السمة";
"Tint_Color" = "لون الصبغة";
"Reset_Tint_Color" = "إعادة تعيين لون الصبغة";
"Auto_Complete_Queue" = "إكمال تلقائي لقائمة الانتظار";
"Enable_Analytics" = "تفعيل الإحصائيات";

"Sileo_Light" = "Sileo فاتح";
"Sileo_Dark" = "Sileo داكن";
"Sileo_Adaptive" = "تكيُّف Sileo";
"System" = "النظام";

"View_Depiction" = "عرض الوصف";

**Review by iA7myd — APPROVED** [review]

**Comment by worldlydev** [comment]
> "Theme" = "السمة";
> "Tint_Color" = "لون الصبغة";
> "Reset_Tint_Color" = "إعادة تعيين لون الصبغة";
> "Auto_Complete_Queue" = "إكمال تلقائي لقائمة الانتظار";
> "Enable_Analytics" = "تفعيل الإحصائيات";
> 
> "Sileo_Light" = "Sileo فاتح";
> "Sileo_Dark" = "Sileo داكن";
> "Sileo_Adaptive" = "تكيُّف Sileo";
> "System" = "النظام";
> 
> "View_Depiction" = "عرض الوصف";

Implemented in https://github.com/Sileo/Sileo/pull/94/commits/5382ebcfd02f1a48d8330b0a356d6cb8337d0f10

**Comment by CharlieWhile13** [comment]
If no new changes are suggested I will merge later today

File: Sileo/Sileo/ar.lproj/Localizable.strings [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+"Download_Progress" = "%@ من %@";
+
+"Queue_Confirm_Button" = "تأكيد";
+"Queue_Clear_Button" = "مسح قائمة الإنتظار";
+
+"Package_Upgrade_Action" = "ترقية";
+"Package_Reinstall_Action" = "إعادة التثبيت";
+"Package_Uninstall_Action" = "إلغاء التثبيت";
+"Package_Modify_Action" = "تعديل";
+"Package_Get_Action" = "جلب";
```
Comment (yazanoo16) [inline Sileo/Sileo/ar.lproj/Localizable.strings:45]: this one could be:
"Package_Get_Action" = "الحصول";

File: Sileo/Sileo/ar.lproj/Localizable.strings [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+"Package_Queue_Action" = "قائمة الإنتظار";
+"Package_Cancel_Action" = "إلغاء";
+
+"Package_Share_Action" = "مشاركة الحزمة...";
+"Package_Developer_Find_Action" = "المزيد بواسطة هذا المطور";
+"Package_Support_Action" = "الدعم";
+"Package_Hold_Enable_Action" = "تجاهل التحديثات المستقبيلة";
```
Comment (yazanoo16) [inline Sileo/Sileo/ar.lproj/Localizable.strings:52]: this one should be:
"Package_Hold_Enable_Action" = "تجاهل التحديثات المستقبلية";

**Review by yazanoo16 — CHANGES_REQUESTED** [review]

File: Sileo/Sileo/ar.lproj/Localizable.strings (hunk 4 above)
Comment (worldlydev) [reply Sileo/Sileo/ar.lproj/Localizable.strings:45]: Fixed

File: Sileo/Sileo/ar.lproj/Localizable.strings (hunk 5 above)
Comment (worldlydev) [reply Sileo/Sileo/ar.lproj/Localizable.strings:52]: Fixed

**Comment by iA7myd** [comment]
No, please keep "Package_Get_Action" = "جلب";
It is the most correct and as translated by Apple

**Comment by worldlydev** [comment]
> No, please keep "Package_Get_Action" = "جلب";
> It is the most correct and as translated by Apple

Reimplemented

**Comment by iA7myd** [comment]
Ok, I think you can go ahead now @CharlieWhile13 
Thanks @worldlydev & @yazanoo16 🌹

**Comment by yazanoo16** [comment]
> No, please keep "Package_Get_Action" = "جلب";
> It is the most correct and as translated by Apple

oh understandable

**Comment by yazanoo16** [comment]
> Ok, I think you can go ahead now @CharlieWhile13
> Thanks @worldlydev & @yazanoo16 🌹

you’re always welcome! i’m so glad for helping here :)

**Pull request merged**